diff --color -ur gcon-clean/configure.ac gcon-workdir/configure.ac
--- gcon-clean/configure.ac	2025-05-19 11:04:13.000000000 +0530
+++ gcon-workdir/configure.ac	2025-05-23 19:02:26.294952269 +0530
@@ -123,7 +123,7 @@
 CPPFLAGS="$PKGCONF_CPPFLAGS $CPPFLAGS"
 LIBS="$LIBS $PKGCONF_LIBS"
 
-AC_CHECK_HEADERS([stdio.h unistd.h pthread.h fcntl.h linux/fb.h sys/ttydefaults.h sys/syscall.h stdlib.h sys/mman.h sys/wait.h sys/ioctl.h termios.h ctype.h stdnoreturn.h pty.h],
+AC_CHECK_HEADERS([stdio.h unistd.h pthread.h fcntl.h linux/fb.h sys/ttydefaults.h stdlib.h sys/mman.h sys/wait.h sys/ioctl.h termios.h ctype.h stdnoreturn.h pty.h],
     [], [AC_MSG_ERROR([required header not found])])
 
 CFLAGS="$OLD_CFLAGS"
diff --color -ur gcon-clean/src/main.c gcon-workdir/src/main.c
--- gcon-clean/src/main.c	2025-05-19 11:04:13.000000000 +0530
+++ gcon-workdir/src/main.c	2025-05-23 19:00:13.044953990 +0530
@@ -22,7 +22,6 @@
 #include <fcntl.h>
 #include <linux/fb.h>
 #include <sys/ttydefaults.h>
-#include <sys/syscall.h>
 #include <flanterm/flanterm.h>
 #include <flanterm/backends/fb.h>
 #include <stdlib.h>
@@ -35,7 +34,7 @@
 #include <stdnoreturn.h>
 #include <pty.h>
 
-static char *const start_path = "/usr/bin/login";
+static char *const start_path = "/usr/bin/bash";
 static char *const args[] = {start_path, NULL};
 
 struct tty_info {
@@ -99,7 +98,7 @@
 
 static void locked_term_write(int tty_idx, const char *msg, size_t len) {
     while (__atomic_test_and_set(&tty_mutex, __ATOMIC_SEQ_CST)) {
-        sched_yield();
+        ;
     }
     flanterm_write(ttys[tty_idx].context, msg, len);
     __atomic_clear(&tty_mutex, __ATOMIC_SEQ_CST);
@@ -133,7 +132,7 @@
 
 static void do_tty_switch(int tty_idx) {
     while (__atomic_test_and_set(&tty_mutex, __ATOMIC_SEQ_CST)) {
-        sched_yield();
+        ;
     }
 
     flanterm_set_autoflush(ttys[current_tty].context, false);
@@ -421,7 +420,7 @@
 
     // Open devices.
     pcspkr = open("/dev/pcspeaker", O_RDWR);
-    kb = open("/dev/ps2keyboard", O_RDONLY);
+    kb = open("/dev/keyboard", O_RDONLY);
     if (kb == -1) {
         perror("Could not open keyboard");
         return 1;
@@ -457,7 +456,7 @@
     termios.c_iflag = BRKINT | IGNPAR | ICRNL | IXON | IMAXBEL;
     termios.c_oflag = OPOST | ONLCR;
     termios.c_cflag = CS8 | CREAD;
-    termios.c_lflag = ISIG | ICANON | ECHO | ECHOE | ECHOK | ECHOCTL | ECHOKE;
+    termios.c_lflag = ISIG | ICANON | ECHO | ECHOE | ECHOK;
     termios.c_cc[VINTR] = CTRL('C');
     termios.c_cc[VEOF] = CTRL('D');
     termios.c_cc[VSUSP] = CTRL('Z');
