diff --color -ur gcon-clean/src/main.c gcon-workdir/src/main.c
--- gcon-clean/src/main.c	2026-03-17 23:07:34.000000000 +0000
+++ gcon-workdir/src/main.c	2026-04-09 15:42:19.229533021 +0000
@@ -37,7 +37,7 @@
 #include <poll.h>
 #include <errno.h>
 
-static char *const start_path = "/usr/bin/login";
+static char *const start_path = "/usr/bin/bash";
 static char *const args[] = {start_path, NULL};
 
 struct tty_info {
@@ -429,8 +429,8 @@
     // to see which one opens.
     // USB keyboards dont have gaps in the naming so we can check for the 1st.
     int result;
-    char *keyboard_paths[2] = {"/dev/ps2keyboard", "/dev/usbkeyboard1"};
-    for (int i = 0; i < 2; i++) {
+    char *keyboard_paths[3] = {"/dev/keyboard", "/dev/ps2keyboard", "/dev/usbkeyboard1"};
+    for (int i = 0; i < 3; i++) {
         result = open(keyboard_paths[i], O_RDONLY);
         if (result >= 0) {
             break;
