diff --color -uNr fastfetch-clean/CMakeLists.txt fastfetch-workdir/CMakeLists.txt
--- fastfetch-clean/CMakeLists.txt	2024-12-26 01:37:25.000000000 +0000
+++ fastfetch-workdir/CMakeLists.txt	2026-04-13 16:14:18.840261727 +0000
@@ -30,6 +30,8 @@
     set(DragonFly TRUE CACHE BOOL "..." FORCE)
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "SunOS")
     set(SunOS TRUE CACHE BOOL "..." FORCE)
+elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Polaris")
+    set(Polaris TRUE CACHE BOOL "..." FORCE)
 elseif(NOT APPLE AND NOT WIN32)
     message(FATAL_ERROR "Unsupported platform: ${CMAKE_SYSTEM_NAME}")
 endif()
@@ -1079,6 +1081,85 @@
         src/util/platform/FFPlatform_unix.c
         src/util/binary_linux.c
     )
+elseif(Polaris)
+	list(APPEND LIBFASTFETCH_SRC
+		src/common/dbus.c
+		src/common/io/io_unix.c
+		src/common/netif/netif_linux.c
+		src/common/networking_linux.c
+		src/common/processing_linux.c
+		src/detection/battery/battery_nosupport.c
+		src/detection/bios/bios_nosupport.c
+		src/detection/bluetooth/bluetooth_nosupport.c
+		src/detection/bluetoothradio/bluetoothradio_nosupport.c
+		src/detection/board/board_nosupport.c
+		src/detection/bootmgr/bootmgr_nosupport.c
+		src/detection/brightness/brightness_nosupport.c
+		src/detection/btrfs/btrfs_nosupport.c
+		src/detection/chassis/chassis_nosupport.c
+		src/detection/cpu/cpu_nosupport.c
+		src/detection/cpucache/cpucache_nosupport.c
+		src/detection/cpuusage/cpuusage_nosupport.c
+		src/detection/cursor/cursor_nosupport.c
+		src/detection/disk/disk_nosupport.c
+		src/detection/dns/dns_linux.c
+		src/detection/physicaldisk/physicaldisk_nosupport.c
+		src/detection/physicalmemory/physicalmemory_nosupport.c
+		src/detection/diskio/diskio_nosupport.c
+		src/detection/font/font_nosupport.c
+		src/detection/host/host_nosupport.c
+		src/detection/initsystem/initsystem_linux.c
+		src/detection/keyboard/keyboard_linux.c
+		src/detection/libc/libc_linux.c
+		src/detection/lm/lm_linux.c
+		src/detection/loadavg/loadavg_nosupport.c
+		src/detection/locale/locale_linux.c
+		src/detection/gamepad/gamepad_linux.c
+		src/detection/media/media_linux.c
+		src/detection/memory/memory_linux.c
+		src/detection/mouse/mouse_linux.c
+		src/detection/netio/netio_linux.c
+		src/detection/opengl/opengl_linux.c
+		src/detection/os/os_polaris.c
+		src/detection/packages/packages_linux.c
+		src/detection/poweradapter/poweradapter_linux.c
+		src/detection/processes/processes_linux.c
+		src/detection/gtk_qt/qt.c
+		src/detection/sound/sound_linux.c
+		src/detection/swap/swap_linux.c
+		src/detection/terminalfont/terminalfont_linux.c
+		src/detection/terminalshell/terminalshell_linux.c
+		src/detection/terminalsize/terminalsize_linux.c
+		src/detection/theme/theme_nosupport.c
+		src/detection/tpm/tpm_linux.c
+		src/detection/uptime/uptime_linux.c
+		src/detection/users/users_nosupport.c
+		src/detection/wallpaper/wallpaper_nosupport.c
+		src/detection/wifi/wifi_nosupport.c
+		src/detection/wm/wm_nosupport.c
+		src/detection/de/de_nosupport.c
+		src/detection/wmtheme/wmtheme_nosupport.c
+		src/detection/camera/camera_linux.c
+		src/detection/zpool/zpool_linux.c
+		src/util/platform/FFPlatform_unix.c
+		src/util/binary_linux.c
+		src/detection/displayserver/linux/displayserver_linux.c
+		src/detection/displayserver/linux/drm.c
+		src/detection/displayserver/linux/wayland/wayland.c
+		src/detection/displayserver/linux/wayland/global-output.c
+		src/detection/displayserver/linux/wayland/zwlr-output.c
+		src/detection/displayserver/linux/wayland/kde-output.c
+		src/detection/displayserver/linux/wayland/wlr-output-management-unstable-v1-protocol.c
+		src/detection/displayserver/linux/wayland/kde-output-device-v2-protocol.c
+		src/detection/displayserver/linux/wayland/kde-output-order-v1-protocol.c
+		src/detection/displayserver/linux/wayland/xdg-output-unstable-v1-protocol.c
+		src/detection/displayserver/linux/wmde.c
+		src/detection/displayserver/linux/xcb.c
+		src/detection/displayserver/linux/xlib.c
+		src/detection/gpu/gpu_nosupport.c
+		src/detection/localip/localip_linux.c
+		src/detection/icons/icons_nosupport.c
+	)
 endif()
 
 if(ENABLE_DIRECTX_HEADERS)
@@ -1189,6 +1270,8 @@
     target_compile_definitions(libfastfetch PUBLIC _GNU_SOURCE)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-char-subscripts")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-rpath,/usr/X11R7/lib -Wl,-rpath,/usr/pkg/lib") # ditto
+elseif(Polaris)
+    target_compile_definitions(libfastfetch PUBLIC _GNU_SOURCE _XOPEN_SOURCE _ATFILE_SOURCE)
 endif()
 
 if(FreeBSD OR OpenBSD OR NetBSD)
diff --color -uNr fastfetch-clean/src/data/structure.txt fastfetch-workdir/src/data/structure.txt
--- fastfetch-clean/src/data/structure.txt	2024-12-26 01:37:25.000000000 +0000
+++ fastfetch-workdir/src/data/structure.txt	2026-04-13 16:14:18.850261727 +0000
@@ -1 +1 @@
-Title:Separator:OS:Host:Kernel:Uptime:Packages:Shell:Display:DE:WM:WMTheme:Theme:Icons:Font:Cursor:Terminal:TerminalFont:CPU:GPU:Memory:Swap:Disk:LocalIp:Battery:PowerAdapter:Locale:Break:Colors
+Title:Separator:OS:Host:Kernel:Uptime:Packages:Display:DE:WM:WMTheme:Theme:Icons:Font:Cursor:Terminal:TerminalFont:CPU:GPU:Memory:Swap:Disk:LocalIp:Battery:PowerAdapter:Locale:Break:Colors
diff --color -uNr fastfetch-clean/src/detection/localip/localip_linux.c fastfetch-workdir/src/detection/localip/localip_linux.c
--- fastfetch-clean/src/detection/localip/localip_linux.c	2024-12-26 01:37:25.000000000 +0000
+++ fastfetch-workdir/src/detection/localip/localip_linux.c	2026-04-13 16:14:18.870261727 +0000
@@ -6,7 +6,9 @@
 #include <string.h>
 #include <ctype.h>
 #include <net/if.h>
+#ifndef __polaris__
 #include <ifaddrs.h>
+#endif
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <stdio.h>
@@ -21,6 +23,7 @@
 #if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__APPLE__) || defined(__NetBSD__)
 #include <net/if_media.h>
 #include <net/if_dl.h>
+#elif defined(__polaris__)
 #else
 #include <netpacket/packet.h>
 #endif
@@ -120,6 +123,7 @@
 
 const char* ffDetectLocalIps(const FFLocalIpOptions* options, FFlist* results)
 {
+#ifndef __polaris__
     struct ifaddrs* ifAddrStruct = NULL;
     if(getifaddrs(&ifAddrStruct) < 0)
         return "getifaddrs(&ifAddrStruct) failed";
@@ -202,6 +206,7 @@
                         ptr[0], ptr[1], ptr[2], ptr[3], ptr[4], ptr[5]);
             addNewIp(results, ifa->ifa_name, addressBuffer, -1, isDefaultRoute, flags, false);
         }
+        #elif __polaris__
         #else
         else if (ifa->ifa_addr->sa_family == AF_PACKET)
         {
@@ -631,6 +636,7 @@
             }
         }
     }
+#endif
 
     return NULL;
 }
diff --color -uNr fastfetch-clean/src/detection/os/os_polaris.c fastfetch-workdir/src/detection/os/os_polaris.c
--- fastfetch-clean/src/detection/os/os_polaris.c	1970-01-01 00:00:00.000000000 +0000
+++ fastfetch-workdir/src/detection/os/os_polaris.c	2026-04-13 16:14:18.870261727 +0000
@@ -0,0 +1,7 @@
+#include "os.h"
+#include <stdio.h>
+void ffDetectOSImpl(FFOSResult* os)
+{
+    ffStrbufSetStatic(&os->name, "Polaris");
+    ffStrbufSet(&os->version, &instance.state.platform.sysinfo.release);
+}
diff --color -uNr fastfetch-clean/src/logo/ascii/polaris.txt fastfetch-workdir/src/logo/ascii/polaris.txt
--- fastfetch-clean/src/logo/ascii/polaris.txt	1970-01-01 00:00:00.000000000 +0000
+++ fastfetch-workdir/src/logo/ascii/polaris.txt	2026-04-13 16:14:18.880261727 +0000
@@ -0,0 +1,17 @@
+                                                            
+              .:                                            
+              ++.                                           
+             :*=+==--::....                                 
+            .+.:*.       ..:-=+*+=-::....                   
+            -=  =+.                 ..::-==+-..             
+............+.   *-..................         .-+=-:..      
+.:**-:::::::.    .::::::::**:........:-+**+-..     ..-++:.  
+   .-*-.              .-+=.                  .:-====-.   :-.
+      .+*:.         .++.      ....:=++++++==-:.........:+*=.
+        .=-        -*.     :==-...            .....::----=-.
+        :+.   ..    +-. .-=.                                
+       .*:  :++=+:  .+.=-.                                  
+      .==.=*-.  .:+=.=*.                                    
+      :*#=.        .-*#:                                    
+      -:.            .:=.                                   
+                                                            
\ No newline at end of file
diff --color -uNr fastfetch-clean/src/logo/builtin.c fastfetch-workdir/src/logo/builtin.c
--- fastfetch-clean/src/logo/builtin.c	2024-12-26 01:37:25.000000000 +0000
+++ fastfetch-workdir/src/logo/builtin.c	2026-04-13 16:14:18.880261727 +0000
@@ -3712,6 +3712,14 @@
             FF_COLOR_FG_WHITE,
         },
     },
+    // Polaris
+    {
+        .names = {"Polaris"},
+        .lines = FASTFETCH_DATATEXT_LOGO_POLARIS,
+        .colors = {
+            FF_COLOR_FG_LIGHT_MAGENTA,
+        },
+    },
     // LAST
     {},
 };
diff --color -uNr fastfetch-clean/src/options/general.h fastfetch-workdir/src/options/general.h
--- fastfetch-clean/src/options/general.h	2024-12-26 01:37:25.000000000 +0000
+++ fastfetch-workdir/src/options/general.h	2026-04-13 16:14:18.930261727 +0000
@@ -16,7 +16,7 @@
     bool detectVersion;
 
     // Module options that cannot be put in module option structure
-    #if defined(__linux__) || defined(__FreeBSD__) || defined(__sun) || defined(__OpenBSD__) || defined(__NetBSD__)
+    #if defined(__linux__) || defined(__FreeBSD__) || defined(__sun) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__polaris__)
     FFstrbuf playerName;
     bool escapeBedrock;
     FFDsForceDrmType dsForceDrm;
diff --color -uNr fastfetch-clean/src/util/FFstrbuf.h fastfetch-workdir/src/util/FFstrbuf.h
--- fastfetch-clean/src/util/FFstrbuf.h	2024-12-26 01:37:25.000000000 +0000
+++ fastfetch-workdir/src/util/FFstrbuf.h	2026-04-13 16:14:18.930261727 +0000
@@ -7,6 +7,7 @@
 #include <stdbool.h>
 #include <stdio.h>
 #include <string.h>
+#include <strings.h>
 #include <stdlib.h>
 #include <assert.h>
 
diff --color -uNr fastfetch-clean/src/util/platform/FFPlatform_unix.c fastfetch-workdir/src/util/platform/FFPlatform_unix.c
--- fastfetch-clean/src/util/platform/FFPlatform_unix.c	2024-12-26 01:37:25.000000000 +0000
+++ fastfetch-workdir/src/util/platform/FFPlatform_unix.c	2026-04-13 16:14:18.930261727 +0000
@@ -19,7 +19,7 @@
 static void getExePath(FFPlatform* platform)
 {
     char exePath[PATH_MAX + 1];
-    #ifdef __linux__
+    #if defined(__linux__) || defined(__polaris__)
         ssize_t exePathLen = readlink("/proc/self/exe", exePath, sizeof(exePath) - 1);
         exePath[exePathLen] = '\0';
     #elif defined(__APPLE__)
